### PR TITLE
Support for multiple IP Ranges in Metallb config

### LINKED ISSUE #1319: Metallb doesn't support multiple IP ranges
Hello,

I'm trying to add multiple IP ranges in metallb but I cannot:

```bash
$ echo "192.168.0.105-192.168.0.105,192.168.0.110-192.168.0.111" | microk8s.enable metallb
Enabling MetalLB
You input value (192.168.0.105-192.168.0.105,192.168.0.110-192.168.0.111) is not a valid IP Range
Failed to enable metallb
```

That's because in this line, the REGEX doesn't consider that possibility of having comma separated IP ranges. https://github.com/ubuntu/microk8s/blob/master/microk8s-resources/actions/enable.metallb.sh#L37

I tested modifying the config map of the metallb to add multiple ranges, and it works:

```yaml
apiVersion: v1
data:
  config: |
    address-pools:
    - name: default
      protocol: layer2
      addresses:
      - 192.168.0.105-192.168.0.105
      - 192.168.0.106-192.168.0.110
kind: ConfigMap
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","data":{"config":"address-pools:\n- name: default\n  protocol: layer2\n  addresses:\n  - 192.168.0.105-192.168.0.105\n  - 192.168.0.106-192.168.0.110\n"},"kind":"ConfigMap","metadata":{"annotations":{},"name":"config","namespace":"metallb-system"}}
  creationTimestamp: "2020-06-17T11:41:32Z"
  managedFields:
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:data:
```

And I deployed two k8s charms with deployment:service=loadbalancer

```bash
$ juju status
Model       Controller          Cloud/Region        Version  SLA          Timestamp
controller  microk8s-localhost  microk8s/localhost  2.7.6    unsupported  14:55:52+02:00

App    Version       Status  Scale  Charm  Store  Rev  OS          Address        Notes
asd    squid:latest  active      1  squid  local    2  kubernetes  192.168.0.106  
squid  squid:latest  active      2  squid  local    1  kubernetes  192.168.0.105  

Unit      Workload  Agent  Address     Ports     Message
asd/0*    active    idle   10.1.73.18  3128/TCP  Unit is ready
squid/1*  active    idle   10.1.73.15  3128/TCP  Unit is ready
squid/2   active    idle   10.1.73.16  3128/TCP  Unit is ready
```

If you see, one IP is of the first range, and the other one from the second range.

Please run `microk8s inspect` and attach the generated tarball to this issue.

We appreciate your feedback. Thank you for using microk8s.

### PR DESCRIPTION
**Pull request by siddharths2710**
Fixes #1319

### CODE REVIEW

**Comment by siddharths2710** [comment]
This is the current behaviour of my changes:

```
$ echo "192.168.0.105-192.168.0.105,192.168.0.110-192.168.0.111" | microk8s.enable metallb
Enabling MetalLB
Applying registry manifest
namespace/metallb-system created
podsecuritypolicy.policy/speaker unchanged
serviceaccount/controller created
serviceaccount/speaker created
clusterrole.rbac.authorization.k8s.io/metallb-system:controller unchanged
clusterrole.rbac.authorization.k8s.io/metallb-system:speaker unchanged
role.rbac.authorization.k8s.io/config-watcher created
clusterrolebinding.rbac.authorization.k8s.io/metallb-system:controller unchanged
clusterrolebinding.rbac.authorization.k8s.io/metallb-system:speaker unchanged
rolebinding.rbac.authorization.k8s.io/config-watcher created
daemonset.apps/speaker created
deployment.apps/controller created
configmap/config created
Applying registry manifest
namespace/metallb-system unchanged
podsecuritypolicy.policy/speaker unchanged
serviceaccount/controller unchanged
serviceaccount/speaker unchanged
clusterrole.rbac.authorization.k8s.io/metallb-system:controller unchanged
clusterrole.rbac.authorization.k8s.io/metallb-system:speaker unchanged
role.rbac.authorization.k8s.io/config-watcher unchanged
clusterrolebinding.rbac.authorization.k8s.io/metallb-system:controller unchanged
clusterrolebinding.rbac.authorization.k8s.io/metallb-system:speaker unchanged
rolebinding.rbac.authorization.k8s.io/config-watcher unchanged
daemonset.apps/speaker unchanged
deployment.apps/controller unchanged
configmap/config configured
$
```

@ktsakalozos is there any other test I can do for the check-in?

**Comment by ktsakalozos** [comment]
Hi @siddharths2710 thank you for this work it is very useful.

To write a test you add a validation function in [1] and you call it from the test-addons [2] and test-upgrade [3] units. With these tests we make sure the addon will not break in future releases and will be upgrading from one release to the next without any problems.

Thank you again for this work.

[1] https://github.com/ubuntu/microk8s/blob/master/tests/validators.py
[2] https://github.com/ubuntu/microk8s/blob/master/tests/test-addons.py
[3] https://github.com/ubuntu/microk8s/blob/master/tests/test-upgrade.py

**Comment by balchua** [comment]
When this is done, i'd like to add cidr support addresses. And upgrade metallb too.

**Comment by siddharths2710** [comment]
@ktsakalozos and @balchua I'm not able to narrow down the build error from Travis CI. Could you help me out with this?

**Comment by balchua** [comment]
I think we hit the travis time limit.  There is a work to move to github actions.  For now we can ignore this type of error.

**Comment by siddharths2710** [comment]
Thank you @balchua. Could you review my check-in and tests for any changes that I may need to make?

**Comment by balchua** [comment]
Lgtm

**Comment by ktsakalozos** [comment]
The pytests did not pass for me, see https://pastebin.ubuntu.com/p/ybtYyPFSGY/

File: microk8s-resources/actions/enable.metallb.sh [hunk 1]
```diff
@@ -23,24 +23,30 @@ fi
...
+for ip_range in "${ip_ranges[@]}"
+do
+  if [[ $ip_range =~ $REGEX_IP_RANGE ]]
+  then
+    ip_range_str="${ip_range_str}\n      - ${ip_range}"
+  else
+    echo "Your input value ($ip_range) is not a valid IP Range"
+    exit 1
+  fi
   echo "Applying registry manifest"
```
[the remark this answers is not in the document]
Comment (siddharths2710) [reply microk8s-resources/actions/enable.metallb.sh]: @ktsakalozos Noted. I've made the change accordingly.

**Comment by siddharths2710** [comment]
@ktsakalozos I looked into it, there's an issue with PyYAML in parsing the configmap output ( in parsing the `annotations` field ). I'm also facing problem in handling it in json format in the same field.

As a workaround, I'm doing a direct check of the ip ranges in the command output filtered via `jsonpath`.
Could you re-run pytest again?

**Comment by ktsakalozos** [comment]
LGTM +1, thank you for this work @siddharths2710

**Pull request merged**